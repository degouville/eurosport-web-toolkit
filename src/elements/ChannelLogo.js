import React from 'react';
import PropTypes from 'prop-types';

const ChannelLogo = ({ channel = 'E' }) => {
  return (<div>{channel}</div>)
};

ChannelLogo.propTypes = {
  channel: PropTypes.oneOf(['E', 'E1', 'E2'])
};

export default ChannelLogo;
