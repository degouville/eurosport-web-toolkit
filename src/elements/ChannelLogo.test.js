import React from 'react';
import { shallow } from 'enzyme';
import ChannelLogo from './ChannelLogo';

it('renders E ChannelLogo as a default', () => {
  expect(shallow(<ChannelLogo />)).toMatchSnapshot();
});

it('renders E1 ChannelLogo', () => {
  expect(shallow(<ChannelLogo channel="E1" />)).toMatchSnapshot();
});

it('renders E2 ChannelLogo', () => {
  expect(shallow(<ChannelLogo channel="E2" />)).toMatchSnapshot();
});
