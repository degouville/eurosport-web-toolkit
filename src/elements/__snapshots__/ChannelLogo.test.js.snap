// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders E ChannelLogo as a default 1`] = `
<Styled(baseIcon)
  height="50"
  type="e"
  width={43.5}
/>
`;

exports[`renders E1 ChannelLogo 1`] = `
<Styled(baseIcon)
  height="50"
  type="e1"
  width={71}
/>
`;

exports[`renders E2 ChannelLogo 1`] = `
<Styled(baseIcon)
  height="50"
  type="e2"
  width={83}
/>
`;
